[Wongqingyee] iP

Wongqingyee opened the pull request:
# TaskFlow

> "The way to get started is to quit talking and begin doing :zipper_mouth_face:." - Walt Disney [source](https://dansilvestre.com/productivity-quotes/)

TaskFlow is a text-based personal assistant chatbot designed to help you keep track of various tasks 📄. 

Why choose TaskFlow?

- Its text-based.
- Its easy to learn and even faster to use.
- Its ***FREE***.

## Getting Started

1. Click [here](https://github.com/Wongqingyee/ip/releases/tag/v0.1) to download the jar file. 
2. Launch the app by double clicking.
3. Add your tasks.
4. Let TaskFlow to help keep track on your task.

### Features

- [x]  Managing tasks.
- [X] Managing deadlines.
- [X] Managing events.

**And more coming soon!**

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Duke {
    public static void main(String[] args) {
        Duke duke = new Duke("./data/duke.txt");
        duke.run();
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Wongqingyee committed:
Level-0: Rename, Greet, Exit

Wongqingyee committed:
Level 1: Echo

Wongqingyee committed:
Level 2: Add, List

Wongqingyee committed:
Level 3: Mark as Done

Wongqingyee committed:
Level 4: ToDos, Events, Deadlines

Wongqingyee committed:
A-TextUiTesting: Automated Text UI Testing

Wongqingyee committed:
Level 5:Handle Errors

Wongqingyee committed:
Level 6: Delete

Wongqingyee committed:
A-Enums: Use Enums

Wongqingyee committed:
Level 7: Save

Wongqingyee committed:
Merge branch 'branch-Level-7'

Wongqingyee committed:
Level-8: Dates and Time

Wongqingyee committed:
Level-8: Dates and Time

Wongqingyee committed:
Merge branch 'branch-Level-8'

Wongqingyee committed:
A-MoreOOP: Use more OOP

Wongqingyee committed:
Merge branch 'branch-A-MoreOOP'

Wongqingyee committed:
A-Packages: Organize into packages

Wongqingyee committed:
Merge branch 'branch-A-Packages'

Wongqingyee committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Wongqingyee committed:
A-Gradle: Use Gradle to checkstyle

Wongqingyee committed:
Merge branch 'branch-A-Gradle'

Wongqingyee committed:
A-JUnit: Add JUnit tests for mark, unmark and delete

Wongqingyee committed:
Merge branch 'branch-A-JUnit'

Wongqingyee committed:
A-Jar: Create a JAR file

Wongqingyee committed:
Merge branch 'branch-A-Jar'

Wongqingyee committed:
A-JavaDoc: Adds javadoc for every class.

Wongqingyee committed:
A-CodingStandard: Follow the coding standard

Wongqingyee committed:
Level-9: Find

Wongqingyee committed:
Merge branch 'branch-A-JavaDoc'

Wongqingyee committed:
Merge branch 'branch-A-CodingStandard'

Wongqingyee committed:
Merge branch 'branch-Level-9'

Wongqingyee committed:
A-CheckStyle: Use CheckStyle

Wongqingyee committed:
Merge branch 'branch-A-CheckStyle'

zhekaiii commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,177 @@
+package duke;
+
+import duke.command.*;
```
Avoid using wildcard imports. IntelliJ has a default number (>3 or >5) of classes before they convert your imports into wildcard imports, you can change this in your settings :)

zhekaiii commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,146 @@
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+
+    private Scanner sc;
```
You can name your larger scope variables like this more clearly, e.g. `private Scanner scanner`

Tip: You can press Shift+F6 over variables to rename them and IntelliJ will rename all usages of the variable for you.

zhekaiii commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,146 @@
+    /**
+     * Prints the welcome message when the chatbot starts.
+     */
+    public void welcomeMsg() {
```
Method names should be verbs. Same goes for `goodbyeMsg` below
```suggestion
    public void showWelcomeMsg() {
```

zhekaiii reviewed (commented):
Your code is really clean and it was pretty hard for me to find any coding standard violations. Well done!

zhekaiii commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,177 @@
+                    + "Please enter 'help' command to find out more.");
+        }
+        String description = inputs[1];
+        return new ToDoCommand(description);
```
Your method name is Todo but your class name is ToDo. Good to standardize :)

zhekaiii reviewed (commented):
Oops I was supposed to look out for Code Quality instead of Coding Standard Violations for this review but again, your code is pretty clean and you name your variables/methods pretty well so I also didn't find much to comment on :)

Wongqingyee committed:
Level 10: GUI

Wongqingyee committed:
Merge branch 'branch-Level-10'

jovantanyk commented:
Running ./gradlew run produces the following error. 
```
> Task :run FAILED
Error: Main method not found in class duke.Duke, please define the main method as:
   public static void main(String[] args)
or a JavaFX application class must extend javafx.application.Application

FAILURE: Build failed with an exception.
```
This might be due to your build.gradle setting the mainClass to be duke.Duke instead of duke.Main or duke.Launcher

Wongqingyee committed:
Add assertions when checking pre-conditions and post-conditions
of the private methods.

Add assertions for input validation in the private methods of
Parser class.

Wongqingyee committed:
A-CodeQuality: Improve Code Quality.

Add code comment in Deadline class to explain the code.

Wongqingyee committed:
Merge pull request #1 from Wongqingyee/branch-A-Assertions

Add assertions

Wongqingyee committed:
A-CodeQuality: Improve Code Quality.

Add code comment in Deadline class to explain the code.

Wongqingyee committed:
Merge pull request #3 from Wongqingyee/branch-A-CodeQuality

Improve code quality

Wongqingyee committed:
A-Streams: Use Streams

Use stream to filter the keyword in findTasksByKeyword().

Use IntStream to map the index range from 0 to size of
the list in list().

Wongqingyee committed:
Merge pull request #4 from Wongqingyee/branch-A-Streams

A-Streams: Use Streams

Wongqingyee committed:
Create gradle.yml

Wongqingyee committed:
Merge pull request #5 from Wongqingyee/Wongqingyee-patch-1

Create gradle.yml

Wongqingyee committed:
A-CI: Set Up CI

Wongqingyee committed:
Merge pull request #6 from Wongqingyee/A-CI

A-CI: Set Up CI

jovantanyk commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,88 @@
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
```
Method name could be clearer in accordance to java convention. Perhaps flipDialogBox()

jovantanyk commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,91 @@
+            } catch (IOException e) {
+                throw new DukeException("Error creating a new File: " + e.getMessage());
+            }
+        } else {
```
If the directory doesn't exist it'll throw an error, maybe have a case to account for when directory doesnt exist

Wongqingyee committed:
Add a new line at the end of the files.

Wongqingyee committed:
Add a new line at the end of the files.

jovantanyk commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,91 @@
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
```
It might be a good idea to make Storage a static class instead since filePath is effectively final and doesn't change. That way we dont have to pass an instance of Storage around

Wongqingyee committed:
Add a new line at the end of the Launcher.java,
DialogBox.java, Main.java and MainWindow.java.

Add a whitespace after the comma for the lambda
expression in MarkCommandTest class.

jovantanyk commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+ * Represents a task with a deadline in the Duke chatbot application.
+ * It is a subclass of the Task class.
+ */
+public class Deadline extends Task {
```
Might be a good idea to name subclasses after the parent class so DeadlineTask, EventTask, TodoTask for further clarity

jovantanyk commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+        this.by = by;
+        try {
+            // Set up a DateTimeFormatter to parse the date/time input in the expected format.
+            DateTimeFormatterBuilder builder = new DateTimeFormatterBuilder();
```
Java Constructors should ideally not contain too much business logic. Parsing for LocalDateTime should ideally happen **before** the deadlinetask is created

jovantanyk reviewed (commented):
Overall very good use of OOP and separation of logic from view. Classes and methods are named with consistency and in accordance to convention. Test cases are also thoughtfully crafted. Good job!

Wongqingyee committed:
Implement the archive feature with archive and unarchive command

Wongqingyee committed:
Merge branch 'branch-C-Archive'

Wongqingyee committed:
Remove the isExit and isArchive method which are not being used

Wongqingyee committed:
Update test methods

Wongqingyee committed:
Make the image of the user icon and chatbot icon round

Wongqingyee committed:
Add an icon for the app

Wongqingyee committed:
Handle duplicate tasks when adding the task to the list

Wongqingyee committed:
Add ParserTest to test the commands

Wongqingyee committed:
Add ParserTest to test the commands

Wongqingyee committed:
Merge branch 'A-BetterGui'

Wongqingyee committed:
Merge branch 'A-Personality'

Wongqingyee committed:
Merge tag 'A-MoreErrorHandling'

Wongqingyee committed:
Merge branch 'A-MoreTesting'

Wongqingyee committed:
Add the missing javadocs for methods

Wongqingyee committed:
Add missing javadocs for methods in UI

Wongqingyee committed:
Update DeleteCommandTest

Wongqingyee committed:
Update the add method in TaskList

Wongqingyee committed:
Update README for the user guide

Wongqingyee committed:
Merge branch 'branch-A-UserGuide'

Wongqingyee committed:
Add Ui.png

Wongqingyee committed:
Re-implement the add method in TaskList

Wongqingyee committed:
Override equals method in Task.java
